docker/Dockerfile_txsim

```diff
@@ -1,5 +1,5 @@
 # Stage 1: generate celestia-appd binary
-FROM --platform=$BUILDPLATFORM docker.io/golang:1.22.4-alpine3.19 as builder
+FROM --platform=$BUILDPLATFORM docker.io/golang:1.22.5-alpine3.19 as builder
```

Comment:
We may want to bump the Go version in all go.mods, README, and Dockerfiles in one PR